conda_forge_tick/update_prs.py

```diff
@@ -0,0 +1,180 @@
+            for i, migration in enumerate(prs):
+                pr_json = migration.get("PR", None)
+                # allow for false
+                if pr_json:
+                    if '__lazy_json__' in pr_json:
+                        m = PR_JSON_REGEX.match(pr_json['__lazy_json__'])
+                        if m:
+                            pr_id = int(m.group(1))
+                        else:
+                            pr_id = object()
+                    else:
+                        pr_id = object()
+                    pr_info_ordered[pr_id] = (pr_json, node_id, i)
+
+        for pr_id, v in pr_info_ordered.items():
+            if v:
+                (pr_json, node_id, i) = v
+                future = pool.submit(update_function, ghctx, pr_json, gh, dry_run)
+                futures[future] = (node_id, i, pr_json)
+
+        for f in as_completed(futures):
+            name, i, pr_json = futures[f]
+            try:
+                res = f.result()
+                if res:
+                    succeeded_refresh += 1
+                    pr_json.update(**res)
+                    logger.info(f"Updated json for {name}: {res['id']}")
+            except github3.GitHubError as e:
+                logger.error(f"GITHUB ERROR ON FEEDSTOCK: {name}")
+                failed_refresh += 1
+                if is_github_api_limit_reached(e, gh):
+                    break
+            except github3.exceptions.ConnectionError:
+                logger.error(f"GITHUB ERROR ON FEEDSTOCK: {name}")
+                failed_refresh += 1
+            except Exception:
+                logger.critical(
+                    "ERROR ON FEEDSTOCK: {}: {}".format(
+                        name, gx.nodes[name]["payload"]["PRed"][i],
+                    ),
+                )
+                raise
+    return succeeded_refresh, failed_refresh
+
+
+def update_graph_pr_status(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
+    succeeded_refresh, failed_refresh = _update_pr(refresh_pr, dry_run, gx)
+
+    logger.info(f"JSON Refresh failed for {failed_refresh} PRs")
+    logger.info(f"JSON Refresh succeed for {succeeded_refresh} PRs")
+    return gx
+
+
+def close_labels(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
+    succeeded_refresh, failed_refresh = _update_pr(close_out_labels, dry_run, gx)
+
+    logger.info(f"bot re-run failed for {failed_refresh} PRs")
+    logger.info(f"bot re-run succeed for {succeeded_refresh} PRs")
+    return gx
+
+
+def main(args: "CLIArgs") -> None:
+    setup_logger(logger)
+
+    if os.path.exists("graph.json"):
+        gx = load_graph()
+    else:
+        gx = None
+    # Utility flag for testing -- we don't need to always update GH
+    no_github_fetch = os.environ.get("CONDA_FORGE_TICK_NO_GITHUB_REQUESTS")
+    if not no_github_fetch:
+        gx = close_labels(gx, args.dry_run)
+        gx = update_graph_pr_status(gx, args.dry_run)
+
+    logger.info("writing out file")
+    dump_graph(gx)
```

Comment:
This is problematic. Both this job and graph making will dump to the same data sources. This will create git merge conflicts and issues.